catch/unit/occupancy/hipOccupancyMaxActiveBlocksPerMultiprocessor.cc

```diff
@@ -16,76 +16,105 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
-#include <hip_test_common.hh>
-
-static __global__ void f1(float *a) { *a = 1.0; }
+/*
+Testcase Scenarios :
+Unit_hipOccupancyMaxActiveBlocksPerMultiprocessor_Positive_RangeValidation - Test correct execution
+of hipOccupancyMaxActiveBlocksPerMultiprocessor for diffrent parameter values
+Unit_hipOccupancyMaxActiveBlocksPerMultiprocessor_Positive_TemplateInvocation - Test correct
+execution of hipOccupancyMaxActiveBlocksPerMultiprocessor template for diffrent parameter values
+Unit_hipOccupancyMaxActiveBlocksPerMultiprocessor_Negative_Parameters - Test unsuccessful execution
+of hipOccupancyMaxActiveBlocksPerMultiprocessor api when parameters are invalid
+*/
+#include "occupancy_common.hh"
 
-template <typename T>
-static __global__ void f2(T *a) { *a = 1; }
+static __global__ void f1(float* a) { *a = 1.0; }
 
-/**
- * Defines
- */
-#define OccupancyDisableCachingOverride 0x01
+template <typename T> static __global__ void f2(T* a) { *a = 1; }
 
-TEST_CASE("Unit_hipOccupancyMaxActiveBlocksPerMultiprocessor_Negative") {
-  hipError_t ret;
-  int numBlock = 0, blockSize = 0;
-  int gridSize = 0, defBlkSize = 32;
+TEST_CASE("Unit_hipOccupancyMaxActiveBlocksPerMultiprocessor_Negative_Parameters") {
+  int numBlocks = 0;
+  int blockSize = 0;
+  int gridSize = 0;
 
   // Get potential blocksize
   HIP_CHECK(hipOccupancyMaxPotentialBlockSize(&gridSize, &blockSize, f1, 0, 0));
 
-  // Validate each argument
-  ret = hipOccupancyMaxActiveBlocksPerMultiprocessor(NULL, f1, blockSize, 0);
-  REQUIRE(ret != hipSuccess);
-
-  ret = hipOccupancyMaxActiveBlocksPerMultiprocessor(&numBlock, NULL, blockSize, 0);
-  REQUIRE(ret != hipSuccess);
-
-  ret = hipOccupancyMaxActiveBlocksPerMultiprocessor(&numBlock, f1, 0, 0);
-  REQUIRE(ret != hipSuccess);
-
-  ret = hipOccupancyMaxActiveBlocksPerMultiprocessor(&numBlock, f1, 0,
-                                         std::numeric_limits<std::size_t>::max());
-  REQUIRE(ret != hipSuccess);
-
-  ret = hipOccupancyMaxActiveBlocksPerMultiprocessorWithFlags(&numBlock, f1,
-                                  defBlkSize, 0, OccupancyDisableCachingOverride);
-  REQUIRE(ret == hipSuccess);
+  // Common negative tests
+  MaxActiveBlocksPerMultiprocessorNegative(
+      [](int* numBlocks, int blockSize, size_t dynSharedMemPerBlk) {
+        return hipOccupancyMaxActiveBlocksPerMultiprocessor(numBlocks, f1, blockSize,
+                                                            dynSharedMemPerBlk);
+      },
+      blockSize);
+
+#if HT_NVIDIA  // EXSWHTEC-219
```

Comment:
remove this and use json file. I talked to @mangupta and we just want to avoid the whole test case.

Reply:
@chrispaquot It has been removed and test has been disabled using json files.